### PR TITLE
Support a directory with protos as input in gRPC tool

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1626

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java [hunk 1]
```diff
@@ -130,14 +128,40 @@ public void execute() {
...
+                outStream.println(errorMessage);
+            } else {
+                for (File protoFile : protoFiles) {
+                    // check input protobuf file path
+                    Optional<String> pathExtension = getFileExtension(protoFile.getPath());
+                    if (!PROTO_EXTENSION.equalsIgnoreCase(pathExtension.get())) {
+                        String errorMessage = "Invalid proto file path: " + protoFile.getPath() +
+                                ". Please input valid proto file location.";
+                        outStream.println(errorMessage);
+                        return;
```
Comment (daneshk) [inline grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java:146]: here we are restricting to have only proto files in the directory. instead, shall we filter only proto files and proceed.

Comment (daneshk) [reply grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java:146]: Shall we also test the scenario where we have a dependent proto file also in the same directory. want to make sure we do not regenerate it again.

File: grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java [hunk 2]
```diff
@@ -74,15 +74,13 @@
...
-    private static final String WHITESPACE_CHARACTOR = " ";
-    private static final String WHITESPACE_REPLACEMENT = "\\ ";
 
     private CommandLine parentCmdParser;
 
     @CommandLine.Option(names = {"-h", "--help"}, hidden = true)
     private boolean helpFlag;
 
-    @CommandLine.Option(names = {"--input"}, description = "Input .proto file")
+    @CommandLine.Option(names = {"--input"}, description = "Input .proto file(s)")
```
Comment (BuddhiWathsala) [inline grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java:83]: Here, the description `files(s)` seems like it allows to add a list of files. But here we need a directory as input. Shall we add a description, something like:

> Input proto file or a directory containing multiple proto files

File: native/src/main/java/io/ballerina/stdlib/grpc/protobuf/cmd/GrpcCmd.java [hunk 3]
```diff
@@ -118,14 +119,56 @@ public void execute() {
...
+                return;
+            }
+            generateBalFile(protoPath);
+        }
+    }
+
+    private List<String> getProtoFiles(Path path) throws IOException {
+        if (!Files.isDirectory(path)) {
+            throw new IllegalArgumentException("Path must be a directory!");
         }
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/grpc/protobuf/cmd/GrpcCmd.java:159]: Do we need to check this here. as we already check before

Comment (MadhukaHarith92) [reply native/src/main/java/io/ballerina/stdlib/grpc/protobuf/cmd/GrpcCmd.java]: Removed in 714b52c

File: grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java (hunk 2 above)
Comment (MadhukaHarith92) [reply grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java]: Updated in 6486d05

File: grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java (hunk 1 above)
Comment (MadhukaHarith92) [reply grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java]: > here we are restricting to have only proto files in the directory. instead, shall we filter only proto files and proceed.

Updated in 832d112


> Shall we also test the scenario where we have a dependent proto file also in the same directory. want to make sure we do not regenerate it again.

Added a test case with dependent proto in a4372b1

**Review by daneshk — APPROVED** [review]

File: ballerina/tests/resources/proto-files/proto-dir/helloWorldWithDependency.proto [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (BuddhiWathsala) [inline ballerina/tests/resources/proto-files/proto-dir/helloWorldWithDependency.proto:2]:
```suggestion
 *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: ballerina/tests/resources/proto-files/proto-dir/message.proto [hunk 5]
```diff
@@ -0,0 +1,26 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (BuddhiWathsala) [inline ballerina/tests/resources/proto-files/proto-dir/message.proto:2]:
```suggestion
 *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**